cli/upload/upload.go

```diff
@@ -47,19 +48,28 @@ func NewCommand() *cobra.Command {
 		Long:    "Upload Arduino sketches. This does NOT compile the sketch prior to upload.",
 		Example: "  " + os.Args[0] + " upload /home/user/Arduino/MySketch",
 		Args:    cobra.MaximumNArgs(1),
+		PreRun:  checkFlagsConflicts,
 		Run:     run,
 	}
 
 	uploadCommand.Flags().StringVarP(&fqbn, "fqbn", "b", "", "Fully Qualified Board Name, e.g.: arduino:avr:uno")
 	uploadCommand.Flags().StringVarP(&port, "port", "p", "", "Upload port, e.g.: COM10 or /dev/ttyACM0")
 	uploadCommand.Flags().StringVarP(&importDir, "input-dir", "", "", "Directory containing binaries to upload.")
+	uploadCommand.Flags().StringVarP(&importFile, "input-file", "i", "", "Binary file to upload.")
 	uploadCommand.Flags().BoolVarP(&verify, "verify", "t", false, "Verify uploaded binary after the upload.")
 	uploadCommand.Flags().BoolVarP(&verbose, "verbose", "v", false, "Optional, turns on verbose mode.")
 	uploadCommand.Flags().StringVarP(&programmer, "programmer", "P", "", "Optional, use the specified programmer to upload or 'list' to list supported programmers.")
 
 	return uploadCommand
 }
 
+func checkFlagsConflicts(command *cobra.Command, args []string) {
```

Comment:
Since we have this check here at lower level https://github.com/arduino/arduino-cli/pull/777/files#diff-2cb7c2b612feb221e1a0e4c63af8ba4bR456

Is this function redundant?

Reply:
The error message from the lower level is:
`Error: retrieving build artifacts: importFile and importDir cannot be used together`

that is ok-ish but the message in the preprun function is:
`error: --input-file and --input-dir flags cannot be used together`

that is much more clear in the command line context, so I will keep it even if redundant.